src/worker/cloudfront.rs

```diff
@@ -0,0 +1,74 @@
+use std::time::SystemTime;
+
+use aws_sigv4::{
+    http_request::{self, SignableRequest, SigningSettings},
+    SigningParams,
+};
+use reqwest::blocking::Client;
+
+#[derive(Clone)]
+pub struct CloudFront {
+    distribution_id: String,
+    access_key: String,
+    secret_key: String,
+}
+
+impl CloudFront {
+    pub fn from_environment() -> Option<Self> {
+        let distribution_id = dotenv::var("CLOUDFRONT_DISTRIBUTION").ok()?;
+        let access_key =
+            dotenv::var("CLOUDFRONT_ACCESS_KEY").expect("missing CLOUDFRONT_ACCESS_KEY");
+        let secret_key =
+            dotenv::var("CLOUDFRONT_SECRET_KEY").expect("missing CLOUDFRONT_SECRET_KEY");
+        Some(Self {
+            distribution_id,
+            access_key,
+            secret_key,
+        })
+    }
+
+    /// Invalidate a file on CloudFront
+    ///
+    /// `path` is the path to the file to invalidate, such as `config.json`, or `re/ge/regex`
```

Comment:
We may want to add a `let path = path.strip_prefix('/').unwrap_or(path);` to handle the case where an initial `/` is included.